src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -103,25 +105,27 @@ public HttpConnection(Uri url, ITransportFactory transportFactory, ILoggerFactor
 
         private Task StartAsyncCore()
         {
-            if (Interlocked.CompareExchange(ref _connectionState, ConnectionState.Connecting, ConnectionState.Initial)
-                != ConnectionState.Initial)
+            if (ChangeState(from: ConnectionState.Disconnected, to: ConnectionState.Connecting) != ConnectionState.Disconnected)
             {
                 return Task.FromException(
-                    new InvalidOperationException("Cannot start a connection that is not in the Initial state."));
+                    new InvalidOperationException("Cannot start a connection that is not in the Disconnected state."));
             }
 
+            _startTcs = new TaskCompletionSource<object>(TaskCreationOptions.RunContinuationsAsynchronously);
+            _eventQueue = new TaskQueue();
+
             StartAsyncInternal()
                 .ContinueWith(t =>
                 {
                     if (t.IsFaulted)
                     {
                         _startTcs.SetException(t.Exception.InnerException);
-                        _closedTcs.TrySetException(t.Exception.InnerException);
+                        Closed?.Invoke(t.Exception);
```

Comment:
`InnerException`

Reply:
I actually removed it altogether - the connection has not been started so the Closed event should not be invoked. Note that the user will get exception from StartAsync.